lib/mailbox.c

```diff
@@ -16,25 +16,116 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <rte_log.h>
+#include <rte_lcore.h>
+#include <rte_debug.h>
+#include <rte_malloc.h>
+
 #include "gatekeeper_mailbox.h"
 
-/*
- * TODO Remove this preprocessing directive when there are
- * are calls to these functions; for now, the mailbox is
- * just an example of how to set up a library.
- */
-#if 0
-void
-create_mailbox(void)
+/* XXX Sample parameters, need to be tested for better performance. */
+#define GK_MEM_CACHE_SIZE (64)
+
+struct mailbox *
+create_mailbox(const char *tag,
+	int ele_count, int ele_size, unsigned int lcore_id)
+{
+	int ret;
+	char ring_name[64];
+	char pool_name[64];
+	char mailbox_name[64];
```

Comment:
Given that we're accepting a `tag`, let's be conservative and use 128 instead of 64 in the sizes above.